gemp-module/gemp-stccg-client/src/main/web/js/gemp-022/gameAnimations.js

```diff
@@ -233,7 +233,22 @@ export default class GameAnimations {
                 if (controllerId != null)
                     participantId = controllerId;
 
-                if (zone == "SPACELINE" && participantId != that.game.bottomPlayerId) {
+                // If card is listed as public knowledge in server -> common/filterable/Zone.java
+                // and the card belongs to the opponent, inform UI to invert it.
+                let visible_opponent_zones = [
+                    "TABLE",
+                    "SPACELINE",
+                    "AT_LOCATION",
+                    "SUPPORT",
+                    "ATTACHED",
+                    "STACKED",
+                    "REMOVED",
+                    "PLAY_PILE"
+                ];
```

Comment:
@datanoh Please verify this is the list of card zones we potentially want to be visible on the board and, if it belongs to the opponent, inverted.

Reply:
Visible on the board = yes, except REMOVED is never visible on the board. Also, SUPPORT no longer exists, you can remove any references on the client side to that zone. (I'm removing the references on the server side.)

For displaying it inverted:
- SPACELINE - yep
- TABLE, AT_LOCATION, ATTACHED, STACKED - I don't have a strong preference on these, but they should all invert (or not invert) the same way. Right now, the "attached" and "stacked" implementations don't flip. For example, in your screencap you can see the card titles for bottom player's attached cards, but the attributes for top player's attached cards. If we're going to invert, it'd be nice to invert how "attached" and "stacked" work too. (Not losing sleep on that for MVP though)
- PLAY_PILE - Never used in 1E or 2E, only in Tribbles. Tribbles is a game for 3-8 players, so if we're reflecting the images, it'll be a different angle than 180. For now anything is fine; Tribbles won't be up for MVP.

Reply:
FWIW - the "zones" are obnoxiously designed, where sometimes they have server implications and sometimes they have client implications.

ATTACHED is the worst perpetrator, as it has almost 0 functionality except for the server to tell the client how to display the card. SPACELINE and AT_LOCATION serve a purpose, but there are probably cleaner ways to implement them in the client once we're passing game states back and forth.

I wasn't planning to take these on prior to MVP, but if it helps on your end to be less reliant on them, just say the word and we can knock it out.

Reply:
We'll need to display REMOVED once we get to the [AU: Twilight cards](https://www.trekcc.org/1e/?cardID=7603) but I hope we will have the UI overhauled by then. Taking it out for MVP, along with PLAY_PILE and SUPPORT.

Reply:
> If we're going to invert, it'd be nice to invert how "attached" and "stacked" work too. (Not losing sleep on that for MVP though)
I get what you're saying but I'm going to leave this as-is because the code to position the card up-down-left-right is separate from the flip-180 code. The up-down-left-right code controls the visual "stacking" behavior and that code is confusing, fragile, and completely untested. I will make a note for the Game UI overhaul and tackle it then.

If we find that people complain about seeing the bottoms of opponents' cards while seeing the tops of their cards, we have two options:

1. Tell them to right-click the bottom of the card to see all the details.
2. Make a one line code change to disable this upside-down class, which returns things to "always show everything right side up" like they were before.

Reply:
> We'll need to display REMOVED once we get to the [AU: Twilight cards](https://www.trekcc.org/1e/?cardID=7603) but I hope we will have the UI overhauled by then. Taking it out for MVP, along with PLAY_PILE and SUPPORT.

Can you clarify you're just removing it from this visible_opponent_zones? Idk if there are other references to REMOVED in the client, but if there are, they should probably be kept. You're right that REMOVED will need to be referenced, but I don't think it should ever be visible on the board - even once we get to AU: Twilight. I'd say it should be like hand/discard/draw deck, where there's a clickable box in the corner.

Reply:
> > If we're going to invert, it'd be nice to invert how "attached" and "stacked" work too. (Not losing sleep on that for MVP though)
> 
> I get what you're saying but I'm going to leave this as-is because the code to position the card up-down-left-right is separate from the flip-180 code. The up-down-left-right code controls the visual "stacking" behavior and that code is confusing, fragile, and completely untested. I will make a note for the Game UI overhaul and tackle it then.
> 
> If we find that people complain about seeing the bottoms of opponents' cards while seeing the tops of their cards, we have three options:
> 
>     1. Tell them to right-click the bottom of the card to see all the details.
> 
>     2. Make a one line code change to disable this upside-down class, which returns things to "always show everything right side up" like they were before.
> 
>     3. Make a small code change to only apply this upside-down class to missions and leave the rest of the cards right side up.

Do you have a timeline in mind for the Game UI overhaul vs. 1.0? I'm 100% with you on the criticisms of attached and stacked. I think it'd be helpful to try to address it at least a little for 1.0. There's a lot of the UI riding on how attached and stacked work, so the sooner we untangle that, the better.